[access_log][bugfix]: memory leak when reopen_logs fail. (#19714)
envoyproxy/envoy: docs/root/version_history/current.rst

---------
*Changes expected to improve the state of the world and are unlikely to have negative effects*

> * access_log: fix memory leak when reopen_logs fail. When reopen log fail, flush thread will exit, leads to memory leak with ``flush_buffer_``.Try to reopen accesslog every flush loop instead of exit, and ``doWrite()`` no matter file is open or not to drain buffer.

Reviewer: ```suggestion
* access_log: fix memory leak when reopening an access log fails. Access logs will now try to be reopened on each subsequent flush attempt after a failure.
```
